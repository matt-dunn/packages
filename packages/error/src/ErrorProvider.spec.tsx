import {shallow} from "enzyme";
import {ErrorProvider as Component} from "./ErrorProvider";
import React from "react";

describe("ErrorBoundary", () => {
  it("should render with no error", () => {
    const wrapper = shallow(
      <Component
        value={{
          error: undefined
        }}
      >
        Simple
      </Component>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("should render with error", () => {
    const wrapper = shallow(
      <Component
        value={{
          error: new Error("Mock error")
        }}
      >
        Simple
      </Component>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
