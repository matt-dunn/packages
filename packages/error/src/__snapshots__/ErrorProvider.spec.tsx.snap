// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary should render with error 1`] = `
<ContextProvider
  value={
    Object {
      "error": [Error: Mock error],
    }
  }
>
  Simple
</ContextProvider>
`;

exports[`ErrorBoundary should render with no error 1`] = `
<ContextProvider
  value={
    Object {
      "error": undefined,
    }
  }
>
  Simple
</ContextProvider>
`;
